scripts/validate_container.sh

```diff
@@ -127,7 +127,7 @@ echo "==        C++ Tests        =="
 echo "============================="
 
 tmpFile=$(mktemp)
-for ex in `find examples/ -name '*.cpp'`;
+for ex in `find examples/ -name '*.cpp' | sort`;
```

Comment:
Some of our tests perform "pip install" during the test. That means that later tests can benefit from having packages installed during prior tests. That occurs when we run our .ipynb files with the notebook validation.

Reply:
I will add a comment that says that the `sort` is to make the test order repeatable.

Reply:
Comment added in 343e602a1

Reply:
Yes, every test _should_ have its own set of pip installs. However, if a test doesn't have all the proper `pip install`s in it, I would rather that it repeatably pass or repeatably fail ... not just be randomly different every time.

For example, consider the hypothetical two files:

### file1.ipynb
```python
!pip install mypackage
import mypackage
# use my package
...
```
### file2.ipynb
```python
import mypackage
# use mypackage
```

If `file1.ipynb` were executed before `file2.ipynb`, everything would work just fine. However, if `file2.ipynb` were executed before `file1.ipynb`, then the `import mypackage` would fail. I would rather that either repeatably pass or repeatably fail.

If everything is working correctly, the `sort` simply makes the logs easier to compare across two runs. It is currently very difficult to compare logs from one publishing run to another publishing run. That alone is enough to justify the `sort` change in my opinion.

Reply:
> Also, I think we aren't cleaning the temp directories after every test, right? If that is the case, then the library should be there on the box.

I am not aware of this happening. Which temp directories? Does this `pip uninstall` everything after every test?

(In any case, I still think repeatable test behavior is worthwhile.)